### PR TITLE
X to features and y to labels

### LINKED ISSUE #570: Rename X,y to features, labels
As suggested here: https://github.com/openml/openml-python/pull/547#discussion_r225066702

### PR DESCRIPTION
**Pull request by skubatur**
#### Reference Issue

Fixes #570 

#### What does this PR implement/fix? Explain your changes.
Renamed X to features and y to labels in the following:
1. Function calls in examples
2. docstring in the examples
3. Function return variables
4. Function docstring

#### How should this PR be tested?
1. Run example scripts. 
2. Run unit tests related to dataset class

#### Any other comments?
If it makes sense to change X to data in some cases, I would be happy to do that.

### CODE REVIEW

File: examples/20_basic/simple_datasets_tutorial.py [hunk 1]
```diff
@@ -38,13 +38,13 @@
...
-# X - An array/dataframe where each row represents one example with
+# features - An array/dataframe where each row represents one example with
 # the corresponding feature values.
-# y - the classes for each example
+# labels - the classes for each example
 # categorical_indicator - an array that indicates which feature is categorical
-# attribute_names - the names of the features for the examples (X) and
-# target feature (y)
-X, y, categorical_indicator, attribute_names = dataset.get_data(
+# attribute_names - the names of the features for the examples (features) and
```
Comment (mfeurer) [inline examples/20_basic/simple_datasets_tutorial.py]: It would be good to replace the features in `the names of the features` with attributes?

File: examples/40_paper/2018_neurips_perrone_example.py [hunk 2]
```diff
@@ -154,9 +154,9 @@ def list_categorical_attributes(flow_type='svm'):
 # pre-processing all retrieved evaluations.
 
 eval_df, task_ids, flow_id = fetch_evaluations(run_full=False, flow_type=flow_type)
-X, y = create_table_from_evaluations(eval_df, flow_type=flow_type)
-print(X.head())
-print("Y : ", y[:5])
+features, labels = create_table_from_evaluations(eval_df, flow_type=flow_type)
+print(features.head())
+print("Y : ", labels[:5])
```
Comment (mfeurer) [inline examples/40_paper/2018_neurips_perrone_example.py]: Could you please replace the Y in the print function, too?

File: examples/40_paper/2018_neurips_perrone_example.py [hunk 3]
```diff
@@ -197,12 +197,12 @@ def list_categorical_attributes(flow_type='svm'):
 # Selecting a task for the surrogate
 task_id = task_ids[-1]
 print("Task ID : ", task_id)
-X, y = create_table_from_evaluations(eval_df, task_ids=[task_id], flow_type='svm')
+features, labels = create_table_from_evaluations(eval_df, task_ids=[task_id], flow_type='svm')
 
-model.fit(X, y)
-y_pred = model.predict(X)
+model.fit(features, labels)
+y_pred = model.predict(features)
```
Comment (mfeurer) [inline examples/40_paper/2018_neurips_perrone_example.py]: Could you please rename variables such as `y_pred`, too, for example to `predictions`?

**Comment by skubatur** [comment]
> Thanks a lot for your contribution. I added a few inline comments. Also, it would be great if you could rename the remaining `X` and `y` as you suggested to keep the code as consistent as possible.

Made those changes, too. Let me know if you find some more.

**Comment by PGijsbers** [comment]
We're very sorry we didn't follow up on this. At this time we no longer feel the renaming is necessary. Thanks anyway for your contribution.
